integreat_compass/cms/views/offers/offer_delete_view.py

```diff
@@ -0,0 +1,40 @@
+import logging
+
+from django.contrib import messages
+from django.contrib.auth.mixins import PermissionRequiredMixin
+from django.core.exceptions import PermissionDenied
+from django.urls import reverse_lazy
+from django.utils.translation import gettext as _
+from django.views.generic.edit import DeleteView
+
+from ...models import Offer
+
+logger = logging.getLogger(__name__)
+
+
+class OfferDeleteView(PermissionRequiredMixin, DeleteView):
+    """
+    View showing all of a user's offers
+    """
+
+    model = Offer
+    http_method_names = ["post"]
+    success_url = reverse_lazy("cms:protected:my_offers")
+
+    def get_permission_required(self):
+        offer_instance = Offer.objects.filter(pk=self.request.POST.get("pk")).first()
+        if offer_instance.creator != self.request.user:
```

Comment:
```suggestion
        if self.get_object().creator != self.request.user:
```
Unfortunately, `self.object` is not yet populated at this point in time, but we could use the `self.get_object()` to retrieve it. Does not make a difference in sql queries though.